tests/std/tests/VSO_0000000_vector_algorithms_search_n/test.cpp

```diff
@@ -12,8 +12,49 @@
 
 #include "test_vector_algorithms_support.hpp"
 
+#pragma warning(disable : 4984) // if constexpr is a C++17 language extension
+#ifdef __clang__
+#pragma clang diagnostic ignored "-Wc++17-extensions"
+#endif // __clang__
+
 using namespace std;
 
+template <class UnderlyingIt>
+struct forward_iter_adaptor {
+    using iterator_category = forward_iterator_tag;
+    using reference         = typename UnderlyingIt::reference;
+    using value_type        = typename UnderlyingIt::value_type;
+    using pointer           = typename UnderlyingIt::pointer;
+    using difference_type   = ptrdiff_t;
+
+    constexpr forward_iter_adaptor() : ptr{} {}
+    constexpr explicit forward_iter_adaptor(const UnderlyingIt ptr_) : ptr(ptr_) {}
+
+    constexpr reference operator*() const {
+        return *ptr;
+    }
+    constexpr pointer operator->() const {
+        return ptr;
+    }
+    constexpr forward_iter_adaptor& operator++() {
+        ++ptr;
+        return *this;
+    }
+    forward_iter_adaptor operator++(int) {

```

Comment:
Although it isn't being exercised, `constexpr` is missing here. (This may be a relic of the old `constexpr` rule about single statements.)

Reply:
This was deleted in P0202R3_constexpr_algorithm_and_exchange test it was copied from.
The iterator with deleted post-increment was not accepted in ranges algorithm, this made me implement it.